lib/fennec/turn.ex

```diff
@@ -69,4 +73,32 @@ defmodule Fennec.TURN do
       end
     %{turn | channels: [channel | channels]}
   end
+
+  @spec has_channel(t, peer_or_number :: Fennec.address | Format.channel_number)
+    :: {:ok, t, Channel.t} | {:error, t}
+  def has_channel(turn_state, peer_or_number) do
+    now = Fennec.Time.system_time(:second)
+    with {:ok, channel}     <- get_channel(turn_state, peer_or_number),
```

Comment:
I know that this is not part of this review, but since I didn't see this code earlier I'll make this point here: I really, really don't think it's a good idea to traverse all channels every single time the worker receives data from any peer. In the end it can be the reason for slower then expected throughput. The (hash) map would should be much better even thought we would need to keep two of them in-memory due to the fact we have two keys. 

This is however only a note, I don't expect any changes in this regard (at least not in this PR).

Reply:
Yup, that probably requires some performance tests. Anyway, thank for the note, we should keep that in mind.